python/orca/src/bigdl/orca/learn/tf2/spark_runner.py

```diff
@@ -294,14 +296,16 @@ def distributed_train_func(self, data_creator, config, epochs=1, verbose=1,
                             validation_steps=validation_steps,
                             validation_freq=validation_freq)
 
-        if checkpoint:
-            try:
-                if self.rank == 0:
-                    put_local_dir_to_remote(os.path.dirname(replaced_checkpoint_path),
-                                            original_checkpoint_dir)
-            finally:
-                shutil.rmtree(os.path.dirname(replaced_checkpoint_path))
-
+        if callbacks:
+            if checkpoint:
+                try:
+                    if self.rank == 0:
+                        put_local_dir_to_remote(os.path.dirname(replaced_checkpoint_path),
+                                                original_checkpoint_dir)
+                finally:
+                    shutil.rmtree(os.path.dirname(replaced_checkpoint_path))
```

Comment:
If `put_local_dir_to_remote` failed, maybe we should not delete `replaced_checkpoint_path` otherwise user's training result will be lost. 

How about printing a warning stating that there is an error and the checkpoint is located at xxx and then users will have a chance to get them manually.

Reply:
ok